lib/ftu/authentication.dart

```diff
@@ -0,0 +1,56 @@
+import 'dart:async';
+import 'package:firebase_auth/firebase_auth.dart';
+
+//REF https://github.com/dart-lang/linter/issues/311
+
+abstract class BaseAuth {
+  Future<String> signIn(String email, String password);
+
+  Future<String> signUp(String email, String password);
+
+  Future<FirebaseUser> getCurrentUser();
+
+  Future<void> sendEmailVerification();
+
+  Future<void> signOut();
+
+  Future<bool> isEmailVerified();
+}
+
+class Auth implements BaseAuth {
+  final FirebaseAuth _firebaseAuth = FirebaseAuth.instance;
+  @override
+  Future<String> signIn(String email, String password) async {
+    AuthResult result = await _firebaseAuth.signInWithEmailAndPassword(
```

Comment:
does `FirebaseAuth.signInWithEmailAndPassword` hash the passwords? If not we should be adding hashing in a later PR.

Reply:
Yes it is. Look at here https://console.firebase.google.com/project/mentorapp-19fcf/authentication/users